src/Tinfoil/Data/Hash.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DeriveGeneric #-}
+module Tinfoil.Data.Hash(
+    Hash(..)
+  , hexDigest
+) where
+
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Base16 as Base16
+import qualified Data.Text.Encoding as T
+
+import           GHC.Generics (Generic)
+
+import           P
+
+-- | Binary representation of a hash.
+newtype Hash =
+  Hash {
+    unHash :: ByteString
+  } deriving (Eq, Show, Generic)
+
+instance NFData Hash
```

Comment:
@olorin arggg sorry, I meant [deepseq-generics](http://hackage.haskell.org/package/deepseq-generics-0.2.0.0/docs/Control-DeepSeq-Generics.html)

^ this is what got merged in to `deepseq`